vowpalwabbit/array_parameters.h

```diff
@@ -146,7 +142,15 @@ class sparse_parameters
     uint64_t index = i & _weight_mask;
     weight_map::const_iterator iter = _map.find(index);
     if (iter == _map.end())
-      return *default_value;
+    {
```

Comment:
is there a reason this used to not mutate the internal map but now we want it to do so?

Reply:
Previously there was this stashed default_value that was always returned here - the problem with that was that it didn't take into account the fact that the default value is dependent on the index.